updater/aws.go

```diff
@@ -110,6 +128,22 @@ func containsAttribute(attrs []*ecs.Attribute, searchString string) bool {
 	return false
 }
 
+// eachPage defines batch processing boundaries for handling paginated results of API calls.
+func eachPage(inputLen int, size int, fn func(start, stop int) error) (int, error) {
+	pageCount := 0
+	for start := 0; start < inputLen; start += size {
+		stop := start + size
+		if stop > inputLen {
+			stop = inputLen
+		}
+		if err := fn(start, stop); err != nil {
+			return 0, err
+		}
+		pageCount++
+	}
+	return pageCount, nil
+}
+
 // filterAvailableUpdates returns a list of instances that have updates available
 func (u *updater) filterAvailableUpdates(bottlerocketInstances []instance) ([]instance, error) {
```

Comment:
Test is missing for this method. It involves quite complicated logic, it would be good to have test cases for this method as well.

Reply:
Its functionality is already covered in the "Paginated" test cases; which is why I opted to not create a test case specifically for this method. Happy to make one though if we think it's really necessary.

Reply:
Per our offline conversation here is the issue for this: [Issue #100](https://github.com/bottlerocket-os/bottlerocket-ecs-updater/issues/100)